### DISCUSSION TITLE
PxTriangleMesh collision error

### DISCUSSION THREAD

**Discussion by yananZZZZ** [NVIDIA-Omniverse/PhysX#181] [Q&A]
### triangle mesh collide with triangle mesh
Hi, 
As I tried to make two mesh shape collide with each others, I got error reported in the GPU dll:

`PxgCudaSolverCore.cpp (1702) : internal error : GPU mComputeBodiesAverageVelocitiesFunction fail to launch kernel!!`
`PxgCudaSolverCore.cpp (1681) : internal error : GPU solveStaticBlock fail to launch kernel!!`
`PxgCudaUtils.h (75) : internal error : SynchronizeStreams cuEventRecord failed with error 700`
`PxgCudaUtils.h (81) : internal error : SynchronizeStreams cuStreamWaitEvent failed with error 700`

As one of my actor are `PxRigidDynamic` I enable the sdf when cook my mesh shape, something like:
![image](https://github.com/NVIDIA-Omniverse/PhysX/assets/122350362/0146e20d-7a00-43e9-a36c-3b995f48b2c8)

since GPU dll is not open sourced, and I have no idea how to debug this kind of issue as mesh vs mesh is a common case in our application, hope someone can help to give some advice here.
I also export the ovd file for this scene maybe it a bit helpful to reproduce the issue.
![image](https://github.com/NVIDIA-Omniverse/PhysX/assets/122350362/b1307d14-30f7-4ccf-9cbe-1590cae96be0)

[MeshCollisionPVD.zip](https://github.com/NVIDIA-Omniverse/PhysX/files/12354444/MeshCollisionPVD.zip)

**Comment by preist-nvidia** [comment]
Tracking this as an issue: https://github.com/NVIDIA-Omniverse/PhysX/issues/182

**Comment by nvtw** [comment]
Hello, I had a look at the the ovd capture but I could not find something that is obviously incorrect. It might be worth to try a couple of things to see if it starts to work afterwards:

- Ensure that the sdf spacing leads to a reasonable sdf resolution. An estimate of the resolution will be boundingBoxDiagonal/sdfSpacing. I tested the scene in omniverse and it worked quite well with a target sdf resolution of 500. The options to sparsify (compress) are identical as in your scene.
- Maybe it is worth to try a kinematic rigid collider instead of a static collider. But a kinematic rigid would also require an sdf. But since both parts are identical, the sdf can be shared.

The file I used to experiment is attached. Maybe it helps to copy some settings.

[MeshCollisionTest2.zip](https://github.com/NVIDIA-Omniverse/PhysX/files/12360035/MeshCollisionTest2.zip)
